### PR TITLE
GCS: fix warnings for GCC 6.2.1

### PR DESCRIPTION
**Pull request by mluessi**
Fixes warning that cropped up when moving to GCC 6.2.1

### CODE REVIEW

**Comment by tracernz** [comment]
Weary of modifying upstream sources. http://eigen.tuxfamily.org/index.php?title=3.3 is supposed to take care of it (#664).

**Comment by mlyle** [comment]
ooh, -Wmisleading-indentation

**Comment by tracernz** [comment]
> ooh, -Wmisleading-indentation

is quite nice. Arguments for {} on single liners weakening. :stuck_out_tongue: 

http://developers.redhat.com/blog/2016/02/26/gcc-6-wmisleading-indentation-vs-goto-fail/

**Comment by mluessi** [comment]
@tracernz eigen 3.3 still isn't out. I agree in general about modifying upstream code, but this is pretty minor and non-functional

**Comment by tracernz** [comment]
@mluessi It is, but it's still a ticking timebomb waiting to get somebody when they update the sources. :wink: Can we wrap the pragma around the Eigen include in our own source?

**Comment by mluessi** [comment]
@tracernz there are 3 places in the GCS where we include Eigen. So we would either need to have the pragma at 3 different places or create an additional header file to wrap the Eigen includes. Thoughts?

**Comment by tracernz** [comment]
@mluessi We should see if Eigen 3.3 does indeed fix this particular issue. If it does then we can merge this as-is and all will be well when we upgrade. If not, then we should probably put the fix in our own code so it doesn't get lost with Eigen upgrades. What do you think?

P.S. 3.3 is now released. ;)

**Comment by tracernz** [comment]
I was able to compile #1456 with g++-6, (using custom qmake spec), with no warnings (at least not from GCC).

```
mike@mike-desktop:~/dev/dronin2$ g++-6 --version
g++-6 (Ubuntu 6.2.0-3ubuntu11~16.04) 6.2.0 20160901
Copyright (C) 2016 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

mike@mike-desktop:~/dev/dronin2$ make all_clean
[ ! -d "/home/mike/dev/dronin2/build" ] || rm -rf "/home/mike/dev/dronin2/build"
mike@mike-desktop:~/dev/dronin2$ QT_SPEC=linux-g++6 make -j8 gcs 1>/dev/null
Project MESSAGE: QMAKE_CXX = /usr/bin/ccache g++-6
Project MESSAGE: QMAKE_CC = /usr/bin/ccache gcc-6
/home/mike/dev/dronin2/ground/gcs/src/plugins/boards_brainfpv/brainre1configuration.ui: Warning: Z-order assignment: '' is not a valid widget.
/home/mike/dev/dronin2/ground/gcs/src/plugins/boards_brainfpv/brainre1configuration.ui: Warning: Z-order assignment: '' is not a valid widget.
mike@mike-desktop:~/dev/dronin2$
```

**Comment by tracernz** [comment]
But in release config I see the same as you... Worrying that we have different warning flags for release...

```
mike@mike-desktop:~/dev/dronin2$ make all_clean
[ ! -d "/home/mike/dev/dronin2/build" ] || rm -rf "/home/mike/dev/dronin2/build"
mike@mike-desktop:~/dev/dronin2$ QT_SPEC=linux-g++6 GCS_BUILD_CONF=release make -j8 gcs 1>/dev/null
Project MESSAGE: QMAKE_CXX = g++-6
Project MESSAGE: QMAKE_CC = gcc-6
/home/mike/dev/dronin2/ground/gcs/src/libs/tlmapcontrol/mapwidget/mapline.cpp: In member function ‘virtual void mapcontrol::MapLine::paint(QPainter*, const QStyleOptionGraphicsItem*, QWidget*)’:
/home/mike/dev/dronin2/ground/gcs/src/libs/tlmapcontrol/mapwidget/mapline.cpp:94:5: error: this ‘if’ clause does not guard... [-Werror=misleading-indentation]
     if (line().dy() >= 0)
     ^~
/home/mike/dev/dronin2/ground/gcs/src/libs/tlmapcontrol/mapwidget/mapline.cpp:97:9: note: ...this statement, but the latter is misleadingly indented as if it is guarded by the ‘if’
         QPointF arrowP1 = line().pointAt(0.5) + QPointF(sin(angle + M_PI / 3) * arrowSize,
         ^~~~~~~
cc1plus: all warnings being treated as errors
make[4]: *** [.obj/release-shared/mapline.o] Error 1
make[4]: *** Waiting for unfinished jobs....
make[3]: *** [sub-tlmapcontrol-make_first-ordered] Error 2
make[2]: *** [sub-libs-make_first-ordered] Error 2
make[1]: *** [sub-src-make_first-ordered] Error 2
make: *** [gcs] Error 2
```

**Comment by tracernz** [comment]
With this PR, minus the Eigen modifications, and #1456 on top, I can compile successfully in release config but it did find this:

```
mike@mike-desktop:~/dev/dronin2$ QT_SPEC=linux-g++6 GCS_BUILD_CONF=release make -j8 gcs 1>/dev/null
Project MESSAGE: QMAKE_CXX = g++-6
Project MESSAGE: QMAKE_CC = gcc-6
/home/mike/dev/dronin2/ground/gcs/src/libs/quazip/zip.c: In function ‘zip64local_SearchCentralDir’:
/home/mike/dev/dronin2/ground/gcs/src/libs/quazip/zip.c:530:5: warning: this ‘for’ clause does not guard... [-Wmisleading-indentation]
     for (i=(int)uReadSize-3; (i--)>0;)
     ^~~
/home/mike/dev/dronin2/ground/gcs/src/libs/quazip/zip.c:538:7: note: ...this statement, but the latter is misleadingly indented as if it is guarded by the ‘for’
       if (uPosFound!=0)
       ^~
/home/mike/dev/dronin2/ground/gcs/src/plugins/boards_brainfpv/brainre1configuration.ui: Warning: Z-order assignment: '' is not a valid widget.
/home/mike/dev/dronin2/ground/gcs/src/plugins/boards_brainfpv/brainre1configuration.ui: Warning: Z-order assignment: '' is not a valid widget.
In file included from /home/mike/dev/dronin2/ground/gcs/src/plugins/config/configstabilizationwidget.h:30:0,
                 from /home/mike/dev/dronin2/ground/gcs/src/plugins/config/configstabilizationwidget.cpp:34:
.uic/ui_stabilization.h: In member function ‘void Ui_StabilizationWidget::setupUi(QWidget*)’:
.uic/ui_stabilization.h:337:10: note: variable tracking size limit exceeded with -fvar-tracking-assignments, retrying without
     void setupUi(QWidget *StabilizationWidget)
          ^~~~~~~
```

That quazip one looks a lot like a genuine bug to me...
https://github.com/d-ronin/dRonin/blob/90f3609/ground/gcs/src/libs/quazip/zip.c#L530-L536
Compare to https://github.com/d-ronin/dRonin/blob/90f3609/ground/gcs/src/libs/quazip/zip.c#L593-L601

Second warning just means there is too much stuff in the debug symbol table so it discarded some.

File: ground/gcs/src/libs/eigen/Eigen/src/Core/arch/SSE/PacketMath.h [hunk 1]
```diff
@@ -29,6 +29,10 @@
 #define EIGEN_CACHEFRIENDLY_PRODUCT_THRESHOLD 16
 #endif
 
+#if defined __GNUC__ && __GNUC__>=6
```
Comment (tracernz) [inline ground/gcs/src/libs/eigen/Eigen/src/Core/arch/SSE/PacketMath.h]: Don't need this with #1456, please remove. :smile:

**Review by tracernz — CHANGES_REQUESTED** [review]
Please remove Eigen bit, and consider addressing quazip issue.

**Comment by mluessi** [comment]
@tracernz removed the Eigen change and added quazip. Should be good

File: ground/gcs/src/libs/eigen/Eigen/src/Core/arch/SSE/PacketMath.h (hunk 1 above)
Comment (mluessi) [reply ground/gcs/src/libs/eigen/Eigen/src/Core/arch/SSE/PacketMath.h]: Done

**Review by tracernz — APPROVED** [review]

**Comment by tracernz** [comment]
Great, thanks! 😁

**Comment by mlyle** [comment]
jenkins, build artifacts please

**Comment by mlyle** [comment]
woops should have merged the other stuff first.. will respin after this build.

**Comment by mlyle** [comment]
jenkins, build artifacts please

**Comment by tracernz** [comment]
LGTM

**Pull request merged**